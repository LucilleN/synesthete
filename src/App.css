.App {
  /* margin-left: 5rem;
  margin-right: 5rem; */
  width: 100%;
  /* padding-left: 5rem;
  padding-right: 5rem; */
}
