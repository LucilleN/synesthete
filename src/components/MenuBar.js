import React, { useState } from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { Redirect } from 'react-router-dom';

const styles = theme => ({
  bar: {
    background: theme.palette.dark.pink,
    height: '60px',
    textAlign: 'center',
    // opacity: '0.8',
    zIndex: 10
  },
  button: {
    height: '100%',
    '&:hover': {
       background: theme.palette.light.pink,
    },
  },
  textNotCurrentPage: {
    color: theme.palette.light.pink,
    fontSize: '1.5rem',
    padding: 10,
    '&:hover': {
       color: theme.palette.dark.purple
    },
  },
  // textCurrentPage: {
  //   color: theme.palette.light.pink,
  //   '&:hover': {
  //      color: theme.palette.dark.purple
  //   },
  // },
  logo: {
    color: theme.palette.white,
    display: 'flex',
    fontSize: '2.25rem',
    paddingLeft: 20,
  }
})

const MenuBar = props => {
  const [homeSelected, setHomeSelected] = useState(false)


  const handleHomeSelect = event => {
    console.log("handleHomeSelect called")
    event.preventDefault()
    setHomeSelected(true)
    console.log(`homeSelected is ${homeSelected}`)
  }

  const { classes } = props

  if (homeSelected) {
    // setHomeSelected(false)
    return (
      <Redirect to={'/'} />
    )
  }

  return (
    <Grid container className={classes.bar}>
      <Grid item xs={3} onClick={handleHomeSelect}>
        <Typography className={classes.logo}>
          synesthete
        </Typography>
      </Grid>
      <Grid item xs={2} className={classes.button} onClick={handleHomeSelect}>
        <Typography className={classes.textNotCurrentPage}>
          home
        </Typography>
      </Grid>
      <Grid item xs={2} className={classes.button}>
        <Typography className={classes.textNotCurrentPage}>
          search
        </Typography>
      </Grid>
      <Grid item xs={2} className={classes.button}>
        <Typography className={classes.textNotCurrentPage}>
          upload
        </Typography>
      </Grid>
      <Grid item xs={2} className={classes.button}>
        <Typography className={classes.textNotCurrentPage}>
          about
        </Typography>
      </Grid>
    </Grid>
  )
}

export default withStyles(styles)(MenuBar)
