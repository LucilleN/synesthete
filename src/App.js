import React, { useEffect } from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { withStyles } from '@material-ui/core/styles'

import Homepage from './pages/Homepage'
import About from './pages/About'
import Search from './pages/Search'
import UploadVisualization from './pages/UploadVisualization'
import Visualization from './pages/Visualization'
import MenuBar from './components/MenuBar'

import { apiHost } from './api'

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#c23063', // magenta
      dark: '#180142' // dark purple
    },
    type: 'dark',
    dark: {
      pink: '#bf265c',
      purple: '#1c004f',
    },
    medium: {
      purple: '#5a0b66',
    },
    light: {
      pink: '#db91aa',
      purple: '#ac91db'
    },
    white: '#FFFFFF',
  },
  typography: {
    fontFamily: ['Dosis', 'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'].join(',')
  }
})

export const defaultErrorText = 'Sorry, but something went wrong.'

const styles = () => ({
  root: {
    width: '100vw',
    height: '100%',
    minHeight: '100vh',
    background: `linear-gradient(0deg, ${theme.palette.dark.pink} 0%, ${theme.palette.dark.purple} 80%)`
  }
})

const routes = (
  <Router>
    <div>
      <MenuBar />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route exact path="/about" component={About} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/upload" component={UploadVisualization} />
        <Route exact path="/visualization" component={Visualization} />
        <Route exact path="/visualization/:songID" component={Visualization} />
      </Switch>
    </div>
  </Router>
)

const App = (props) => {

  useEffect(() => apiHost('https://api.spotify.com/v1/'))

  const { classes } = props

  return (
    <div className={classes.root}>
      <MuiThemeProvider theme={theme}>
        {routes}
      </MuiThemeProvider>
    </div>
  )
}

export default withStyles(styles)(App)
