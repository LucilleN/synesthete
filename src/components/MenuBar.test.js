import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

import { MuiThemeProvider } from '@material-ui/core/styles'

import { BrowserRouter as Router } from 'react-router-dom'

import { theme } from '../App'
import MenuBar from './MenuBar'

let component
beforeEach(() => {
  component = TestRenderer.create(
    <Router>
      <MuiThemeProvider theme={theme}>
        <MenuBar />
      </MuiThemeProvider>
    </Router>,
  )
})

describe('initial state', () => {
  it('should render the Synesthete logo', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should have four navigation buttons', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('clicking on the logo and on each of the buttons', () => {
  let div
  let clickableItems
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <Router>
          <MuiThemeProvider theme={theme}>
            <MenuBar />
          </MuiThemeProvider>
        </Router>,
        div,
      )
    })
    const logo = div.querySelector('#logo')
    const home = div.querySelector('#home')
    const search = div.querySelector('#search')
    const upload = div.querySelector('#upload')
    const about = div.querySelector('#about')
    clickableItems = [logo, home, search, upload, about]
  })

  afterEach(() => ReactDOM.unmountComponentAtNode(div))

  it('should still render a menu bar after the redirect', () => {
    clickableItems.forEach(clickableItem => {
      ReactTestUtils.act(() => {
        ReactTestUtils.Simulate.click(clickableItem)
      })
      const menuBar = div.querySelector('#bar')
      expect(menuBar).not.toBeNull()
    })
  })
})
