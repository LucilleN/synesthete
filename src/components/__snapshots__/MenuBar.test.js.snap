// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state should have four navigation buttons 1`] = `
<div>
  <div
    className="MuiGrid-root MenuBar-bar-1 MuiGrid-container"
    id="bar"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      id="logo"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-logo-4 MuiTypography-body1"
      >
        synesthete
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="home"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        home
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="search"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        search
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="upload"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        upload
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="about"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        about
      </p>
    </div>
  </div>
</div>
`;

exports[`initial state should render the Synesthete logo 1`] = `
<div>
  <div
    className="MuiGrid-root MenuBar-bar-1 MuiGrid-container"
    id="bar"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      id="logo"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-logo-4 MuiTypography-body1"
      >
        synesthete
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="home"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        home
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="search"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        search
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="upload"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        upload
      </p>
    </div>
    <div
      className="MuiGrid-root MenuBar-button-2 MuiGrid-item MuiGrid-grid-xs-2"
      id="about"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MenuBar-textNotCurrentPage-3 MuiTypography-body1"
      >
        about
      </p>
    </div>
  </div>
</div>
`;
